integration/appaccess/pack.go

```diff
@@ -672,17 +674,25 @@ func (p *Pack) startRootAppServers(t *testing.T, count int, extraApps []service.
 	require.NoError(t, err)
 	require.Equal(t, len(configs), len(servers))
 
-	for _, appServer := range servers {
+	for i, appServer := range servers {
 		srv := appServer
 		t.Cleanup(func() {
 			require.NoError(t, srv.Close())
 		})
-		waitAppServerTunnel(t, p.rootCluster.Tunnel, p.rootAppClusterName, srv.Config.HostUUID)
+		waitForAppServer(t, p.rootCluster.Tunnel, p.rootAppClusterName, srv.Config.HostUUID, configs[i].Apps.Apps)
 	}
 
 	return servers
 }
 
+func waitForAppServer(t *testing.T, tunnel reversetunnel.Server, name string, hostUUID string, apps []service.App) {
+	waitAppServerTunnel(t, tunnel, name, hostUUID)
```

Comment:
Is this check still necessary? Could we rely on having the entry on the cache to ensure the server is ready to receive calls?

I'm asking because it seems that without this check, it will be closer to what would happen in a production environment.

Reply:
Removed, We can relay only on `waitForAppRegInRemoteSiteCache` check.